code/game/objects/gamemode/crash/machines.dm

```diff
@@ -54,6 +54,15 @@
 	current_timer = null
 	visible_message("<b>[src]</b> shuts down as it loses power. Any running programs will now exit")
 
+
+/obj/machinery/computer/nuke_disc_generator/attackby(obj/item/I, mob/user, params)
+	. = ..()
+
+	if(isxeno(user))
```

Comment:
Xenos can't get an attackby since they will never have something in their hands. Also you should use /living/ for the argument.